app/components/dashboard/replication_status_component.html.erb

```diff
@@ -2,33 +2,21 @@
   <div class="card-body" />
     <div class="card-title fs-4 fw-semibold d-flex justify-content-between">
       Replication Zips
-      <span class="badge rounded-pill <%= replication_and_zip_parts_ok? ? 'bg-success' : 'bg-danger' %>">
-        <%= replication_and_zip_parts_ok? ? 'OK' : 'Error' %>
+      <span class="badge rounded-pill <%= replication_zips_badge_class %>">
+        <%= replication_zips_status_label %>
       </span>
     </div>
     <ul class="list-group border border-2 border-secondary">
       <li class="list-group-item d-flex justify-content-between align-items-center">
         <h6 class="d-flex justify-content-start">ZipPart Statuses</h6>
-        <span class="badge rounded-pill <%= zip_parts_ok? ? 'bg-success' : 'bg-danger' %>">
-          <%= zip_parts_ok? ? 'OK' : 'Error' %>
+        <span class="badge rounded-pill <%= zip_parts_badge_class %>">
+          <%= zip_parts_status_label %>
         </span>
       </li>
-      <% endpoint_data.each do |endpoint_name, endpoint_info| %>
-        <% replication_count = endpoint_info[:replication_count] %>
-        <% endpoint_replication_ok = replication_count == num_object_versions_per_preserved_object %>
-        <li class="list-group-item d-flex justify-content-between align-items-center">
-          <h6 class="d-flex justify-content-start pe-2">
-            <span class="text-secondary pe-1">Endpoint:</span>
-            <%= endpoint_name %>
-          </h6>
-          <span class="badge rounded-pill <%= endpoint_replication_ok ? 'bg-success' : 'bg-danger' %>">
-            <%= endpoint_replication_ok ? 'OK' : 'Error' %>
-          </span>
-        </li>
-      <% end %>
+      <%= render Dashboard::ReplicationEndpointStatusComponent.with_collection(endpoints) %>
```

Comment:
Prefer using `ViewComponent#with_collection` over looping and rendering each collection item using Ruby iteration.